components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/choreo/ChoreoStatisticsPublisher.java

```diff
@@ -0,0 +1,293 @@
+        analyticsEnabledForEndpoints = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.ENDPOINT_ANALYTICS_ENABLED, true);
+        analyticsEnabledForInboundEndpoints = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.INBOUND_ENDPOINT_ANALYTICS_ENABLED, true);
+        analyticsEnabled = SynapsePropertiesLoader.getBooleanProperty(
+                ChoreoConstants.SynapseConfigKeys.ANALYTICS_ENABLED, false);
+    }
+
+    protected ChoreoStatisticsPublisher() {
+        loadConfigurations();
+    }
+
+    /**
+     * Process Method of data publisher.
+     *
+     * In this,
+     * It will be checked if analytics is enabled by configuration.
+     * Then, it will be checked if the artifact's analytics is disabled by configuration.
+     * Then, if artifact is Endpoint or Sequence, it will be checked if the artifact is null.
+     *
+     * @param publishingFlow The publishing flow
+     * @param tenantId       The tenant id
+     */
+    @Override
+    public void process(PublishingFlow publishingFlow, int tenantId) {
+        if (!analyticsEnabled) {
+            return;
+        }
+        if (publishingFlow.getEvents().toArray().length > 0) {
+            publishingFlow.getEvents().forEach(event -> {
+                if (event.getElasticMetadata() == null || !event.getElasticMetadata().isValid()) {
+                    return;
+                }
+                if ((StatisticsConstants.FLOW_STATISTICS_API.equals(event.getComponentType()) &&
+                                analyticsEnabledForAPI) ||
+                        (StatisticsConstants.FLOW_STATISTICS_SEQUENCE.equals(event.getComponentType()) &&
+                                analyticsEnabledForSequences) ||
+                        (StatisticsConstants.FLOW_STATISTICS_ENDPOINT.equals(event.getComponentType()) &&
+                                analyticsEnabledForEndpoints) ||
+                        (StatisticsConstants.FLOW_STATISTICS_INBOUNDENDPOINT.equals(event.getComponentType()) &&
+                                analyticsEnabledForInboundEndpoints) ||
+                        (StatisticsConstants.FLOW_STATISTICS_PROXYSERVICE.equals(event.getComponentType()) &&
+                                analyticsEnabledForProxyServices)) {
+                    if ((StatisticsConstants.FLOW_STATISTICS_SEQUENCE.equals(event.getComponentType()) &&
+                            event.getElasticMetadata().getSequence(event.getComponentName()) == null) ||
+                            (StatisticsConstants.FLOW_STATISTICS_ENDPOINT.equals(event.getComponentType()) &&
+                            event.getElasticMetadata().getEndpoint(event.getComponentName()) == null)) {
+                        // If Endpoint or Sequence is null
+                        return;
+                    }
+                    publishAnalytics(event);
+                }
```

Comment:
Shall we do early exits? 
Something like:
for eg:
```java
                if ((StatisticsConstants.FLOW_STATISTICS_API.equals(event.getComponentType()) &&
                                !analyticsEnabledForAPI) {
                    return;
                }
                // Similarly for PROXY and INBOUND_ENDPOINT

                if ((StatisticsConstants. FLOW_STATISTICS_SEQUENCE.equals(event.getComponentType()) {
                    if (!analyticsEnabledForAPI || 
                             event.getElasticMetadata().getSequence(event.getComponentName()) == null)) {
                        return;
                    }
                }
                // Similarly for ENDPOINT

                publishAnalytics(event);      
                
```

Reply:
Changed by https://github.com/wso2/micro-integrator/pull/2927/commits/7ce08a2dbe17617040a9584d9afe40c25a4178fc.

Since we have to publishing Analytics for only 5 artefacts, `publishAnalytics(event)` added inside each if cases.